### PR TITLE
acl2: build with ccl

### PR DESCRIPTION
**Pull request by tani**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

I am original author of this formula.
Current ACL2 bottled by Homebrew does not work well because of the specs of SBCL.
I propose to build ACL2 with Clozure CL. I've already tested the bottle in my local computer.
I hope it works well for all computer too. Thanks.

### CODE REVIEW

**Comment by SMillerDev** [comment]
Does upstream recommend this?

**Comment by tani** [comment]
According to the official [documentation](https://www.cs.utexas.edu/users/moore/acl2/v8-1/HTML/installation/requirements.html), ACL2 team suuports the following LISP compilers.

- Allegro Common Lisp
- CCL (OpenMCL)
- CLISP
- CMU Common Lisp
- GCL
- LispWorks
- SBCL 

At this time, we can install SBCL, CLISP, and SBCL with Homebrew. 
ACL2 team does not recommend to compile source with CLISP. 
Thus, we should build it with SBCL or CCL. MacPorts also provide two options, SBCL and CCL, to build ACL2. 

https://github.com/macports/macports-ports/blob/749ad2787c30497020bdc4cca116eff9f09c800b/math/acl2/Portfile

In previous formula, we built acl2 with ACL2_PAR option. We also can do it with CCL.

File: Formula/acl2.rb [hunk 1]
```diff
@@ -4,45 +4,45 @@ class Acl2 < Formula
...
+  depends_on "openssl@1.1"
+  depends_on "z3"
 
   def install
+    suffix =
+      if OS.mac?
+        "dx86cl64"
+      elsif OS.linux?
+        "lx86cl64"
+      end
```
Comment (SMillerDev) [inline Formula/acl2.rb:25]: @iMichka do we have a better way here?

Comment (iMichka) [reply Formula/acl2.rb:25]: I mean; ask upstream to make their stuff easier with a plain "make" / "make install" step, as I find it weird that we have to go through so many hoops here to get stuff done. This is all prone to future breakage.

But this should do it:
```
suffix = nil
on_macos do
  suffix = "dx86cl64"
end
on_linux do
  suffix = "lx86cl64"
end
```

Comment (tani) [reply Formula/acl2.rb]: Okay, I update this formula to run acl2 without the OS condition.
Could you review it again?

File: Formula/acl2.rb [hunk 2]
```diff
@@ -4,45 +4,46 @@ class Acl2 < Formula
...
-    (bin/"acl2p").write <<~EOF
-      #!/bin/sh
-      export ACL2_SYSTEM_BOOKS='#{libexec}/books'
-      #{Formula["sbcl"].opt_bin}/sbcl --core '#{libexec}/saved_acl2p.core' --userinit /dev/null --eval '(acl2::sbcl-restart)'
+      IMAGE=`find #{libexec} -name 'saved_acl2pr.*' | head -n1`
+      if [ -z "$IMAGE" ]
+      then
+        echo "ACL2 image is not found in #{libexec}" 1>&2
+        exit 1
+      else
```
Comment (SMillerDev) [inline Formula/acl2.rb:36]: This should really be up to upstream to define and maintain

Comment (tani) [reply Formula/acl2.rb]: I agree with you. I will transfer this issue and ask improvement of installation to them.
I remove launcher script. As you said, we should not use wrapper script for future updates.

Next, I'd like to discuss the solution to run bottle made from current formulae.

**Comment by tani** [comment]
Thank you for taking your time. Please read bellow.

We cannot use current bottles in any operating system.
SBCL dumps the binary image(*) with system information, thus the acl2 raises error if the information is different from build environment. So we should remove the bottle from the formula if we keep to use SBCL for ACL2.

On the other hand, we can switch to use Clozure CL to build ACL2 like the switching from GCC to Clang.
I prefer to this solution because we can use bottles. I feel it is natural to fix the formula if the bottle is broken, right?

**Pull request closed without merging**